homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,335 @@
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.RateLimitExceededException as err:
+            _LOGGER.error("RateLimitExceededException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.BadAttributeException as err:
+            _LOGGER.error("BadAttributeException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.TwoFactorException as err:
+            _LOGGER.error("TwoFactorException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.GithubException as err:
+            _LOGGER.error("GithubException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+
+        if self.setup_error is True:
+            return
+
+        if CONF_NAME in repository:
+            self.name = repository[CONF_NAME]
+        else:
+            self.name = repo.name
+
+        self.update = Throttle(interval)(self._update)
+
+        self.stargazers = None
+        self.topics = None
+        self.branches = None
+        self.open_issues = None
+        self.open_pull_requests = None
+        self.last_commit = None
+        self.latest_release = None
+        self.available = False
+
+    def _update(self):
+        """Update GitHub Sensor."""
+        try:
+            repo = self._github_obj.get_repo(self.repository_path)
+
+            self.stargazers = repo.stargazers_count
+
+            self.topics = json.dumps(repo.get_topics())
+
+            branches = []
+            for branch in repo.get_branches():
+                branches.append(branch.name)
+            self.branches = json.dumps(branches)
+
+            open_issues = []
+            for issue in repo.get_issues(state='open', sort='created'):
+                if issue.pull_request is None:
+                    labels = []
+                    for label in issue.labels:
+                        labels.append(label.name)
+                    open_issues.append({
+                        "number": issue.number,
+                        "title": issue.title,
+                        "url": issue.html_url,
+                        "body": issue.body,
+                        "labels": labels,
+                        "assignee": issue.assignee,
+                        "milestone": issue.milestone,
+                        "comments": issue.comments,
+                        "user": {
+                            "login": issue.user.login,
+                            "name": issue.user.name,
+                            "avatar_url": issue.user.avatar_url,
+                            "url": issue.user.html_url
+                        }
+                    })
+            self.open_issues = json.dumps(open_issues)
```

Comment:
This is way too much information, all this stuff gets stored in the recorder at every change. Maybe a count would be enough?

Reply:
How about a count and storing just the latest issue? That way you can send a notification etc. with the issue details